examples/usb/host/src/host.c

```diff
@@ -40,27 +36,27 @@ int main(void)
 
   printf("started\n");
 
-  while (!done) {
-    for (i = 0; !done && i < 4; i++) {
-      printf("Enter number of LEDs to be lit: ");
-      int num_leds;
-      scanf("%d", &num_leds);
-      payload[0] = (unsigned char)num_leds;
-      if (control_write_command(RESOURCE_ID, CONTROL_CMD_SET_WRITE(0), payload, 1) != CONTROL_SUCCESS) {
-        printf("control write command failed\n");
-        exit(1);
-      }
-      fflush(stdout);
+  for (i = 0; i < 4; i++) {
+    payload[0] = i;
+    if (control_write_command(RESOURCE_ID, CONTROL_CMD_SET_WRITE(0), payload, 1) != CONTROL_SUCCESS) {
+      printf("control write command failed\n");
+      exit(1);
+    }
+    fflush(stdout);
+
+    pause_short();
 
-      pause_short();
+    if (control_read_command(RESOURCE_ID, CONTROL_CMD_SET_READ(0), payload, 1) != CONTROL_SUCCESS) {
+      printf("control read command failed\n");
+      exit(1);
+    }
 
-      if (control_read_command(RESOURCE_ID, CONTROL_CMD_SET_READ(0), payload, 2) != CONTROL_SUCCESS) {
-        printf("control read command failed\n");
-        exit(1);
-      }
-      printf("Last button event: %c, value: %d\n", 'A' + payload[0], payload[1]);
-      fflush(stdout);
+    if (payload[0] != i) {
+      printf("control read command returned the wrong value, expected %d, returned %d\n", i, payload[0]);
+      exit(1);
     }
+    printf("Written and read back command with payload: 0x%02X\n", payload[0]);
+    fflush(stdout);
```

Comment:
is this needed?

Reply:
I removed all the calls to fflush(), as they are not used after error printouts.